### PR TITLE
Mitochondrial pipeline - Mity

### PR DESCRIPTION
**Pull request by anjalijain22**

### CODE REVIEW

File: rules/mito_variants.smk [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        "mitochondrial_variants/{family}.mity.vcf.gz"
+    output:
+        "mitochondrial_variants/vcfanno/{family}.mity.vcfanno.vcf"
+    log:
+        "logs/mity/vcfanno/{family}.mity.vcfanno.log"
+    threads: 10
+    resources:
+        mem_mb = 20000
+    params:
+       	conf ="~/crg2/vcfanno/mt.vcfanno.conf",
```
Comment (Madelinehazel) [inline rules/mito_variants.smk:34]: Can you specify the vcfanno as a config parameter instead please? E.g. like [this](https://github.com/ccmbioinfo/crg2/blob/mity/config.yaml#L84).

File: wrappers/mity/call/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+log = snakemake.log_fmt_shell(stdout=True, stderr=True)
+
+family = snakemake.wildcards.family
+outdir = snakemake.params.outdir
+prefix = snakemake.params.prefix
+tool = snakemake.params.tool
+print(snakemake.input.bam)
+
+mod1= (" module load python; ")
+mod2= (" module load freebayes/1.3.1; ")
```
Comment (Madelinehazel) [inline wrappers/mity/call/wrapper.py:13]: Why do we need to load python and freebayes modules if they are present in the environment.yaml?

File: wrappers/mity/report/wrapper.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import process_report
+
+log = snakemake.log_fmt_shell(stdout=True, stderr=True)
+
+family = snakemake.wildcards.family
+outdir = snakemake.params.outdir
+prefix = snakemake.params.prefix
+tool = snakemake.params.tool
+
+mod1 = " module load python; "
```
Comment (Madelinehazel) [inline wrappers/mity/report/wrapper.py:12]: Same comment as above- we shouldn't be loading modules here as they are system specific. All tools we use in a rule should be loaded into the conda environment, or specified via a path in the config.yaml.

File: wrappers/mity/report/environment.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  - numpy==1.16.4
+  - pandas==1.0.0
+  - pysam==0.15.3
+  - python-dateutil==2.8.0
+  - pytz==2019.1
+  - pyvcf==0.6.8
+  - scipy==1.3.2
+  - six==1.12.0
+  - xlsxwriter==1.1.8
+  - freebayes ==1.3.1
```
Comment (Madelinehazel) [inline wrappers/mity/report/environment.yaml:15]: Do we need all these packages/tools to generate the report?

**Review by Madelinehazel — CHANGES_REQUESTED** [review]
Looks good! Just have a few comments regarding the vcfanno config and the environments.

File: wrappers/mity/call/wrapper.py (hunk 2 above)
Comment (anjalijain22) [reply wrappers/mity/call/wrapper.py:13]: Note: Discussed and agreed to keep this the way it is for now.

File: rules/mito_variants.smk (hunk 1 above)
Comment (anjalijain22) [reply rules/mito_variants.smk]: Done!

File: Snakefile [hunk 5]
```diff
@@ -62,8 +68,13 @@ elif config["run"]["pipeline"] == "wgs":
     include: base + "sv.smk"
     include: base + "svreport.smk"
     include: base + "str.smk"
+    include: base + "mito_variants.smk"
 elif config["run"]["pipeline"] == "annot":
     base = "rules/"
+elif config["run"]["pipeline"] == "mity":
+    base = "rules/"
+    include: base + "mito_variants.smk"
```
Comment (Madelinehazel) [inline Snakefile:76]: This also needs to include `base + "mapping.smk"` if the intent is to run the genome from scratch

File: rules/mito_variants.smk [hunk 6]
```diff
@@ -0,0 +1,63 @@
+import pandas as pd
+from datetime import datetime
+
+configfile: "config.yaml"
```
Comment (Madelinehazel) [inline rules/mito_variants.smk:4]: Is this necessary here? In the future we'll have two configs (one for hpf, the other for CHEO-RI), so I don't want this hard-coded.

**Review by Madelinehazel — COMMENTED** [review]
Just some minor comments- doing some test runs, so might have more to add once those are done.

File: Snakefile (hunk 5 above)
Comment (anjalijain22) [reply Snakefile:76]: you're right, I'll add that in!

File: rules/mito_variants.smk (hunk 6 above)
Comment (anjalijain22) [reply rules/mito_variants.smk:4]: That's a good point! I was initially running into some issues (can't remember what it was exactly) which is why I added this in, but I can look more into this.

**Pull request merged**